superset/assets/javascripts/explore/components/AdhocMetricEditPopoverTitle.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormControl, OverlayTrigger, Tooltip } from 'react-bootstrap';
+import AdhocMetric from '../AdhocMetric';
+
+const propTypes = {
+  adhocMetric: PropTypes.instanceOf(AdhocMetric),
+  onChange: PropTypes.func.isRequired,
+};
+
+export default class AdhocMetricEditPopoverTitle extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isHovered: false,
+      isEditable: false,
+    };
+  }
+
+  render() {
+    const { adhocMetric, onChange } = this.props;
+
+    const editPrompt = <Tooltip id="edit-metric-label-tooltip">Click to edit label</Tooltip>;
+
+    return (
+      <OverlayTrigger
+        placement="top"
+        overlay={editPrompt}
+        onMouseOver={() => this.setState({ isHovered: true })}
```

Comment:
it's better to not define functions in the `render` method because you initialize a new value every render. ideally they are bound to the component (or someday we'll enable class-level arrow functions). there's precedent for render functions throughout superset that we can clean up in one go sometime, so maybe you don't have to update, just FYI.